src/main/java/com/epam/controller/LoginController.java

```diff
@@ -0,0 +1,53 @@
+package com.epam.controller;
+
+import com.epam.dao.UserDao;
+import com.epam.model.User;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Optional;
+
+@RestController
+@RequiredArgsConstructor
+public class LoginController {
+    private final UserDao userDao;
+
+    @GetMapping("/login")
+    public ResponseEntity<String> login(@RequestParam String username, @RequestParam String password) {
+        Optional<User> optionalUser = userDao.findByUsername(username);
+        if (optionalUser.isPresent()) {
+            User user = optionalUser.get();
+            if (user.getPassword().equals(password)) {
+                return ResponseEntity.ok("Login successful. Status: 200 OK");
```

Comment:
We need to store the logged user here - this is how the manual authentication is supposed to work here. And all other actions on your REST API should be allowed for the user, who did previously log in. Otherwise you'll need to check username and password match on **each** API call in **each** Rest Controller. See the point 3 at the Notes section.

Reply:
This is still not done, as I can understand. You need to mark user as logged in somehow. And then you need to check on each call like `getSomethingFromMyService(String something, User user)`:
```
if (user.isLoggedIn()) {
 return findSomethingInDaoForString(string);
}
throw new UserIsNotLoggedInException();

Reply:
You can store logged users in some Service, which could have a simple list with user names or Ids, who passed the checkPassword process. I mean that you do not need to store this information in DB. And after logout you'll just delete a user from this list of logged in users.